### PR TITLE
Merge fields with interface while parsing args too, and other improvements

### LINKED ISSUE #20: Add another default arg for `order`
The current default order arg, $order, adds the args `orderBy` and `orderDirection` but we have mostly used `sort` and `order` args for these.

Add another default arg, maybe $sort, that uses those 2 names for the args.

### LINKED ISSUE #22: Add an enum map of GqlEnum instances for easier use
Something like `$gql.enums.DocType.Product` which is a GqlEnum class instance with with Enum name as value

### PR DESCRIPTION
**Pull request by rohit-gohri**
- Fix interface fields were not being considered in `$default` in args
- Add `graphql-list-fields` and `includesField` function
- Closes #20
- Closes #22

### CODE REVIEW

**Comment by rohit-gohri** [comment]
@hit12369

File: src/Gql.js [hunk 1]
```diff
@@ -236,15 +237,29 @@ class Gql {
...
+
+	get fragments() {
+		return this._fragments;
 	}
 
 	static toGqlArg = toGqlArg;
 
+	toGqlArg = toGqlArg;
+
+	args(arg, {pick} = {}) {
```
Comment (ajeet-smpx) [inline src/Gql.js]: Method name should be `arg` as defined in typescript. And it does not handle array type `opts`.

File: src/index.js [hunk 2]
```diff
@@ -12,17 +13,26 @@ async function generateTypesFromSchema(graphqlSchemas, {contextType = 'any', out
 	catch (err) {
 		throw new Error('You need to install \'typescript\' as a dependency');
 	}
-
 	const folder = outputPath || `${process.cwd()}/typings/graphql`;
 
 	schema = _.castArray(schema);
 
-	const getTypeStringForFragments = (fragments, schemaName) => `\
+	const getTypeStringForFragments = ({fragments, enums}, schemaName) => `\
```
Comment (ajeet-smpx) [inline src/index.js]: better variable name: getTypeStringForFragmentsAndEnums ?

File: src/index.js [hunk 3]
```diff
@@ -12,17 +13,26 @@ async function generateTypesFromSchema(graphqlSchemas, {contextType = 'any', out
...
+
 
+import {GqlEnum, GqlFragment} from 'gqutils';
 
 declare global {
 	namespace GraphQl.${schemaName} {
-		type fragments = '${Object.keys(fragments).join("' | '")}';
+		type fragments = {
+			${Object.keys(fragments).map(key => `${key}: GqlFragment;`).join('\n\t\t\t')}
```
Comment (ajeet-smpx) [inline src/index.js]: can use `name` instead of `key` for better readability

**Review by ajeet-smpx — CHANGES_REQUESTED** [review]

File: src/Gql.js [hunk 4]
```diff
@@ -236,15 +237,33 @@ class Gql {
...
+
+	get fragments() {
+		return this._fragments;
 	}
 
 	static toGqlArg = toGqlArg;
 
+	toGqlArg = toGqlArg;
+
+	arg(arg, opts = []) {
```
Comment (ajeet-smpx) [inline src/Gql.js]: default empty array will pick nothing from `arg`. Can check for empty array in `toGqlArg` or remove default value.

**Review by ajeet-smpx — CHANGES_REQUESTED** [review]

File: src/Gql.js (hunk 4 above)
Comment (rohit-gohri) [reply src/Gql.js]: Oops

**Review by ajeet-smpx — APPROVED** [review]

**Pull request merged**